ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -831,9 +832,58 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: [Positive TxSizeInBytes] -> Property
+prop_TxSeq_splitAfterTxSize xs =
+    conjoin [ splitAndCheck txsz
+            | TxTicket _ _ txsz <- txTickets txseq ]
+  where
+    txSizes :: [TxSizeInBytes]
+    txSizes = map getPositive xs
+
+    txseq :: TxSeq Int
+    txseq = TxSeq.toTxSeq $ zip3 (repeat 0) (repeat (TicketNo 0)) txSizes
+
+    splitAndCheck :: TxSizeInBytes -> Bool
+    splitAndCheck txsz =
+      let (before, _after) = splitAfterTxSize txseq txsz
+      in  checkTxSizeSumLessThanOrEqual (txTickets before) txsz
+
+    txSizeSum :: [TxTicket tx] -> TxSizeInBytes
+    txSizeSum = sum . map txTicketTxSizeInBytes
+
+    checkTxSizeSumLessThanOrEqual :: [TxTicket tx] -> TxSizeInBytes -> Bool
+    checkTxSizeSumLessThanOrEqual txs txsz = txSizeSum txs <= txsz
+
+-- | Test that the results of 'splitAfterTxSizeSpec', a specification of
+-- 'splitAfterTxSize', match those of the real 'splitAfterTxSize'
+-- implementation.
+prop_TxSeq_splitAfterTxSizeSpec :: [Small TxSizeInBytes] -> Property
+prop_TxSeq_splitAfterTxSizeSpec xs =
+    conjoin [ splitAndCheck txsz
+            | TxTicket _ _ txsz <- tickets ]
```

Comment:
Wait a second, are you using the size of each transaction to split on? Then you won't get many interesting splits.

I suggest the following: introduce a data type `TxSizesAndMaxSize` (or something better) which consists of a `[Positive TxSizeInBytes]` and a `TxSizeInBytes`. The latter is generated using `choose (1, txSizes ..)`, so you'll get interesting spits. Let `prop_TxSeq_splitAfterTxSizeSpec` (and the other) take it as its argument. You can add a function from `TxSizesAndMaxSize` to `TxSeq Int`, like your local `txseq`, which you can reuse in both tests.

Reply:
Makes sense. I defined a type `TxSizeSplitTestSetup` (name is meh) that consists of a `[TxSizeInBytes]` (all of the generated tx sizes) and a `TxSizeInBytes` which is the actual size to split on in a test. This size to split on is generated using `choose (0, sumOfGeneratedTxSizes)`.